src/query/storage/promremote/storage.go

```diff
@@ -139,6 +139,13 @@ func (p *promStorage) startAsync() {
 			case <-ticker.C:
 				for tenant, queries := range p.pendingQuery {
 					if len(queries) == 0 {
+						delete(p.pendingQuery, tenant)
+						continue
+					}
+					if len(queries) < 5 {
+						p.logger.Error("pending queries has less than 5 samples, skip it, likely due to corrupt tenant header",
```

Comment:
it doesn't mean the data won't be sent but just wait for more samples to come.

Reply:
Why not say that explicitly in the log message? Otherwise, it will be confusing in a couple years.